source/agora/crypto/Schnorr.d

```diff
@@ -128,6 +128,23 @@ public struct Signature
         const signature = Signature.fromString("0x921405afbfa97813293770efd55865c01055f39ad2a70f2b7a04ac043766a693074360d5eab8e888df07d862c4fc845ebd10b6a6c530919d66221219bba50216");
         assert(sig == signature);
     }
+
+    /// print to hex string
+    string toString() const @safe
+    {
+        import std.format;
+
+        return format!"%s%s"(this.R.toString(), this.s.toString(PrintMode.Clear)[2 .. Signature.sizeof + 2]);
+    }
+
+    ///
+    @safe unittest
+    {
+        const signature = Signature(
+            Point("0x921405afbfa97813293770efd55865c01055f39ad2a70f2b7a04ac043766a693"),
+            Scalar("0x074360d5eab8e888df07d862c4fc845ebd10b6a6c530919d66221219bba50216"));
+        assert(signature.toString() == "0x921405afbfa97813293770efd55865c01055f39ad2a70f2b7a04ac043766a693074360d5eab8e888df07d862c4fc845ebd10b6a6c530919d66221219bba50216", signature.toString());
+    }
```

Comment:
This allocates every time. We should use the `sink` overload instead.

Reply:
After https://github.com/bpfkorea/crypto/pull/25 you can just do:
```D
public void toString (scope void delegate(scope const(char)[]) @safe sink)
        const
{
    this.R.toString(sink);
    FormatSpec!char spec;
    spec.spec = 'x';
    this.s.toString(sink, spec);
}
```

Reply:
Thanks, updated.